src/librustc_mir/const_eval.rs

```diff
@@ -643,14 +643,17 @@ pub fn const_eval_raw_provider<'a, 'tcx>(
         let err = error_to_const_error(&ecx, error);
         // errors in statics are always emitted as fatal errors
         if tcx.is_static(def_id).is_some() {
-            let reported_err = err.report_as_error(ecx.tcx,
-                                                   "could not evaluate static initializer");
             // Ensure that if the above error was either `TooGeneric` or `Reported`
             // an error must be reported.
-            if tcx.sess.err_count() == 0 {
-                tcx.sess.delay_span_bug(err.span,
+            let reported_err = tcx.sess.track_errors(|| {
+                err.report_as_error(ecx.tcx,
+                                    "could not evaluate static initializer");
```

Comment:
The `;` here drops the value you'd get for `v` and makes it `()`.

Removing the `;` here should fix your problem

Reply:
Thank you! I dropped `;` but the compile was failed. However, the error message was changed.

```
error[E0308]: mismatched types==================================>    ] 128/137: rustc_mir
   --> src/librustc_mir/const_eval.rs:657:21
    |
657 |                     v
    |                     ^ expected (), found enum `rustc::mir::interpret::ErrorHandled`
    |
    = note: expected type `()`
               found type `rustc::mir::interpret::ErrorHandled`
```

related code is here:

```Rust

if tcx.is_static(def_id).is_some() {
    // Ensure that if the above error was either `TooGeneric` or `Reported`
    // an error must be reported.
    let reported_err = tcx.sess.track_errors(|| {
        err.report_as_error(ecx.tcx,
                            "could not evaluate static initializer")
    });
    match reported_err {
        Ok(v) => {
            tcx.sess.delay_span_bug(err.span,
                                &format!("static eval failure did not emit an error: {:#?}",
                                v));
            v
        },
        Err(ErrorReported) => Err(ErrorHandled::Reported),
    }
    reported_err
}
```

Reply:
The match isn't returning the values written in it, because you have a trailing expression (`reported_err`) afterwards. Removing that will make the `match` the trailing expression

Reply:
I got it, modified and pushed.